ql/src/java/org/apache/hadoop/hive/ql/Compiler.java

```diff
@@ -100,6 +100,7 @@ public QueryPlan compile(String rawCommand, boolean deferClose) throws CommandPr
     boolean parsed = false;
     QueryPlan plan = null;
     try {
+      SessionState.getSessionConf().set(Context.operationNameConf, org.apache.commons.lang3.StringUtils.EMPTY);
```

Comment:
why do we need this?

Reply:
Done, removed checking of operation type as part of compact operation.